ui/redux/utils/modalReducer.js

```diff
@@ -24,9 +24,10 @@ export const setModalConfirm = (modalName, confirmMessage) => dispatch =>
 
 // root reducer
 export default {
-  modal: createObjectsByIdReducer(UPDATE_MODAL_STATE, null, { open: false, confirming: false, confirmOnClose: null }),
+  modal: createObjectsByIdReducer(UPDATE_MODAL_STATE),
 }
 
 // basic selectors
 export const getModalOpen = (state, modalName) => state.modal[modalName] && state.modal[modalName].open
 export const getModalConfim = (state, modalName) => state.modal[modalName] && (state.modal[modalName].confirming || null) && state.modal[modalName].confirmOnClose
+export const getOpenModals = state => Object.keys(state.modal).filter(modalName => state.modal[modalName].open)
```

Comment:
just revisiting the modal dialogs code & state
 - would rename `open` => `isOpen` and `getModalOpen` => `getModalIsOpen`
 - `confirmOnClose` => `confirmOnClosePanel` (?) or  `confirmOnCloseContent` 
 - `getModalConfim` => `getModalConfirmOnClosePanel` etc.


Also, it would be good to have a standard way to document the purpose of components and props - especially ones where it isn't immediately obvious what they do - for example the "popup" prop in Modal.jsx
There's  https://react-styleguidist.js.org/docs/documenting.html , but we don't have to follow it religiously.